src/diff-snapshot.js

```diff
@@ -13,63 +13,58 @@
  */
 
 const fs = require('fs');
-const BlinkDiff = require('blink-diff');
-const intersection = require('lodash/intersection');
+const PNG = require('pngjs').PNG;
+const pixelmatch = require('pixelmatch');
 const mkdirp = require('mkdirp');
 const path = require('path');
 
-const unsupportedDiffConfigKeys = [
-  'imageAPath',
-  'imageA',
-  'imageBPath',
-  'imageB',
-  'imageOutputPath',
-];
-
-function isDiffConfigValid(customDiffConfig) {
-  return intersection(unsupportedDiffConfigKeys, Object.keys(customDiffConfig)).length === 0;
-}
-
 function diffImageToSnapshot(options) {
   const {
     imageData,
     snapshotIdentifier,
     snapshotsDir,
     updateSnapshot = false,
     customDiffConfig = {},
-   } = options;
-
-  if (!isDiffConfigValid(customDiffConfig)) {
-    throw new Error(
-      `Passing in options: ${unsupportedDiffConfigKeys} via Blink-Diff configuration `
-      + 'is not supported as those option are internally used. '
-      + 'Instead pass your image data as first argument to this function!'
-    );
-  }
+  } = options;
 
   let result = {};
   const baselineSnapshotPath = path.join(snapshotsDir, `${snapshotIdentifier}-snap.png`);
   if (fs.existsSync(baselineSnapshotPath) && !updateSnapshot) {
     const outputDir = path.join(snapshotsDir, '__diff_output__');
     const diffOutputPath = path.join(outputDir, `${snapshotIdentifier}-diff.png`);
     const defaultBlinkDiffConfig = {
-      imageA: imageData,
-      imageBPath: baselineSnapshotPath,
-      thresholdType: 'percent',
       threshold: 0.01,
-      imageOutputPath: diffOutputPath,
     };
 
     mkdirp.sync(outputDir);
     const diffConfig = Object.assign({}, defaultBlinkDiffConfig, customDiffConfig);
-    const diff = new BlinkDiff(diffConfig);
-    const unformattedDiffResult = diff.runSync();
 
-    result = Object.assign(
-      {},
-      unformattedDiffResult,
-      { diffOutputPath }
+    const comparisonImg = PNG.sync.read(imageData);
```

Comment:
What happens if the comparison image dimensions are different from the baseline? Would it gracefully error out or do we need to throw a friendly error?

Reply:
Just fine, no weird errors it just fails the comparison.